File: Java/src/main/java/sofe/exercise3/TreeNode.java
package sofe.exercise3;

public class TreeNode {
    TreeNode left;
    TreeNode right;
    int value;

    public TreeNode(int value) {
        this.value = value;
    }

    public TreeNode(int value, TreeNode left, TreeNode right) {
        this.left = left;
        this.right = right;
        this.value = value;
    }

    @Override
    public String toString() {
        return new StringBuilder().append("TreeNode{").
                append("left=").
                append(left).
                append(", right=").
                append(right).
                append(", value=").
                append(value).
                append('}').
                toString();
    }
}

File: C Sharp/Sofe/Exercise3/TreeNode.cs
using System;

namespace Sofe.Exercise3
{
    public class TreeNode
    {
        public TreeNode Left;
        public TreeNode Right;
        public int Value;

        public TreeNode(int Value, TreeNode Left = null, TreeNode Right = null)
        {
            this.Left = Left;
            this.Right = Right;
            this.Value = Value;
        }
    }
}